From 2212fa65eb37bda419a5c9e8531fe1b39ded26a4 Mon Sep 17 00:00:00 2001
In-Reply-To: <20210114233515.31298-1-avarab@gmail.com>
References: <20210114233515.31298-1-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Sat, 16 Jan 2021 15:56:12 +0100
Subject: [PATCH v2 08/11] upload-pack tests: avoid a non-zero "grep" exit
 status
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Continue changing a test that 763b47bafa (t5703: stop losing return
codes of git commands, 2019-11-27) already refactored. A follow-up
commit will add support for testing under bash's "set -o pipefail",
under that mode this test will fail because sometimes there's no
commits in the "objs" output.

It's easier to just fix this than to exempt these tests under a
soon-to-be added "set -o pipefail" test mode. So let's do that.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 t/t5703-upload-pack-ref-in-want.sh | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/t/t5703-upload-pack-ref-in-want.sh b/t/t5703-upload-pack-ref-in-want.sh
index eab966985b..420cf2bbde 100755
--- a/t/t5703-upload-pack-ref-in-want.sh
+++ b/t/t5703-upload-pack-ref-in-want.sh
@@ -16,7 +16,11 @@ get_actual_commits () {
 	test-tool pkt-line unpack-sideband <out >o.pack &&
 	git index-pack o.pack &&
 	git verify-pack -v o.idx >objs &&
-	grep commit objs | cut -d" " -f1 | sort >actual_commits
+	>actual_commits &&
+	if grep -q commit objs
+	then
+		grep commit objs | cut -d" " -f1 | sort >actual_commits
+	fi
 }
 
 check_output () {
-- 
2.29.2.222.g5d2a92d10f8

